Dockerfile

```diff
@@ -1,46 +1,55 @@
-FROM debian:jessie
+FROM alpine:3.3
 
 MAINTAINER Datadog <package@datadoghq.com>
 
+ENV DD_HOME /opt/datadog-agent
 ENV DOCKER_DD_AGENT yes
 ENV AGENT_VERSION 1:5.8.0-1
 
-# Install the Agent
-RUN echo "deb http://apt.datadoghq.com/ stable main" > /etc/apt/sources.list.d/datadog.list \
- && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C7A7DA52 \
- && apt-get update \
- && apt-get install --no-install-recommends -y datadog-agent="${AGENT_VERSION}" \
- && apt-get clean \
- && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+# prevent the agent from being started after install
+ENV DD_START_AGENT 0
+
+# Add Docker check
+COPY conf.d/docker_daemon.yaml "$DD_HOME/agent/conf.d/docker_daemon.yaml"
+
+# Add install and config files
+ADD https://raw.githubusercontent.com/DataDog/dd-agent/master/packaging/datadog-agent/source/setup_agent.sh /setup_agent.sh
+COPY entrypoint.sh /entrypoint.sh
+
+# Install minimal dependencies
+RUN apk add --update curl-dev python-dev tar sysstat
```

Comment:
more efficient by adding `-U --no-cache` flags
`-U` updated the repo before installing
`--no-cache` disables caching the package sources, which isn't needed for production